br/pkg/lightning/backend/kv/base.go

```diff
@@ -94,6 +95,13 @@ func (row RowArrayMarshaller) MarshalLogArray(encoder zapcore.ArrayEncoder) erro
 				return err
 			}
 		}
+		if len(str) > 512*1024 {
+			str = str[0:1024] + " (truncated)"
+		}
+		totalLength += len(str)
+		if totalLength >= 512*1024 {
+			return nil
```

Comment:
We can also log some information, to let user know we have exited early.

Maybe by `encoder.AppendString()`

Reply:
modified